digi/xbee/reader.py

```diff
@@ -658,14 +671,28 @@ def __try_read_packet(self, operating_mode=OperatingMode.API_MODE):
             xbee_packet[0] = self.__serial_port.read_byte()
             while xbee_packet[0] != SpecialByte.HEADER_BYTE.value:
                 xbee_packet[0] = self.__serial_port.read_byte()
-            packet_length = self.__serial_port.read_bytes(2)
+
             # Add packet length.
-            xbee_packet += packet_length
-            length = utils.length_to_int(packet_length)
-            # Add packet payload and checksum.
-            for _ in range(0, length + 1):
-                self.__read_byte_into_packet(xbee_packet, operating_mode)
-            return xbee_packet
+            packet_length_byte = bytearray()
+            for _ in range(0, 2):
+                packet_length_byte += self.__read_next_byte(operating_mode)
+            xbee_packet += packet_length_byte
+            # Length needs to be un-escaped to obtain its integer equivalent.
+            length = utils.length_to_int(XBeeAPIPacket.unescape_data(packet_length_byte))
```

Comment:
This assumes that the LSB of the length is not going to be the escape byte itself (when not in escaped API mode). What happens if the length field is, say, `01 7D`? When traversing that field, `unescape_data` will consume the `01` but ignore `7D`.